File: backend/src/main/java/net/zorphy/backend/main/dto/game/stats/GameStats.java
package net.zorphy.backend.main.dto.game.stats;

import net.zorphy.backend.main.dto.game.stats.chart.ChartData;
import net.zorphy.backend.main.dto.game.stats.correlation.CorrelationResult;
import net.zorphy.backend.main.dto.game.stats.metrics.GameStatsMetrics;
import net.zorphy.backend.main.dto.game.stats.metrics.GameStatsStreak;
import net.zorphy.backend.main.dto.player.PlayerDetails;

import java.time.Duration;
import java.util.List;


public record GameStats(
        PlayerDetails player,
        int gamesPlayed,
        double winRate,
        GameStatsStreak currentWinStreak,
        GameStatsStreak maxWinStreak,
        GameStatsMetrics<Double> scoreMetrics,
        GameStatsMetrics<Duration> durationMetrics,
        PlayerDetails nemesis,
        PlayerDetails victim,
        PlayerDetails rival,
        PlayerDetails companion,
        List<CorrelationResult> correlations,
        List<ChartData> chartData,
        GameSpecificStats gameSpecific
) {
}

File: frontend/src/app/main/games/dto/stats/GameStats.ts
import {PlayerDetails} from "../../../core/dto/PlayerDetails";
import {CorrelationResult} from "./correlations/CorrelationResult";
import {GameStatsMetrics} from "./metrics/GameStatsMetrics";
import {ChartDataHistory} from "./charts/ChartDataHistory";
import {GameStatsStreakResult} from "./streaks/GameStatsStreakResult";

export interface GameStats {
    player: PlayerDetails;
    gamesPlayed: number;
    winRate: number;
    winStreaks: GameStatsStreakResult;
    scoreMetrics: GameStatsMetrics<number>,
    durationMetrics: GameStatsMetrics<string>,
    nemesis: PlayerDetails | null;
    victim: PlayerDetails | null;
    rival: PlayerDetails | null;
    companion: PlayerDetails | null;
    correlations: CorrelationResult[];
    chartData: ChartDataHistory[];
    gameSpecific: any;
}